File: 20180714/Maximize Distance to Cloest Person.py
class Solution(object):
    def maxDistToClosest(self, seats):
        """
        :type seats: List[int]
        :rtype: int
        """
        dis = 0
        cur_dis = 0
        for i in range(len(seats)):
            if seats[i] == 0:
                cur_dis += 1
            else:
                if i - cur_dis - 1 < 0:
                    dis = max(dis, cur_dis)
                if cur_dis % 2 == 0:
                    dis = max(dis, cur_dis / 2)
                else:
                    dis = max(dis, cur_dis / 2 + 1)
                cur_dis = 0
        if cur_dis > dis:
            return cur_dis
        else:
            return dis
        

File: 20180714/Maximize Distance to Cloest Person.java
class Solution {
    public int maxDistToClosest(int[] seats) {
        int count=0;
        int max=Integer.MIN_VALUE;
        int start=0;
        int flag=0;
        for(int i=0; i<seats.length; i++) {
            if(seats[i]==0) {
                count++;
                if(flag==0) start++;
            }
            else{
                flag=1;
                max=Math.max(max,count);
                count=0;
            }
        }
        max=(max%2==0)?max/2:max/2+1;
        return Math.max(start, Math.max(max, count));
    }
}
